src/hist/plot.py

```diff
@@ -210,185 +220,386 @@ def plot2d_full(
     return main_art, top_art, side_art
 
 
+def _construct_gaussian_callable(
+    _hist: hist.BaseHist,
+) -> Callable[[np.ndarray, float, float, float], np.ndarray]:
+    x_values = _hist.axes[0].centers
+    hist_values = _hist.values()
+
+    # gaussian with reasonable initial guesses for parameters
+    constant = float(hist_values.max())
+    mean = (hist_values * x_values).sum() / hist_values.sum()
+    sigma = (hist_values * np.square(x_values - mean)).sum() / hist_values.sum()
+
+    def gauss(
```

Comment:
This can/should be pulled out; it does not need to be a closure. Also the return is np.ndarray, since x is an np.ndarray.

Reply:
> This can/should be pulled out; it does not need to be a closure

Maybe I'm missing the obvious but then should we get rid of this `_construct_gaussian_callable` function and just have these lines lower down in `_plot_ratiolike`? I guess I don't see how to avoid a closure as we want all model parameters of `gauss` to be set now and then get `x` later as `xdata` in `_curve_fit_wrapper`.

> the return is np.ndarray, since x is an np.ndarray.

This is what I thought as well and then was confused by `mypy`'s response

```
src/hist/plot.py:245: error: Returning Any from function declared to return "ndarray"
```

:?

Reply:
That's because a UFunc returns Any, because following though the possible return types is hard. But you know you have an ndarray. You can force it with `retval: np.ndarray = <ufunc>; return retval`, or a `# type: ignore`.

Reply:
I might have missed that it really was a closure, that's fine if it is then.

Reply:
Ahh, it's a closure due to the default arguments.

Reply:
I added a note and forced the return type like you suggested. :+1: